libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookup.java

```diff
@@ -10,14 +10,28 @@
 package org.elasticsearch.entitlement.runtime.policy;
 
 import java.nio.file.Path;
-import java.util.function.Function;
 import java.util.stream.Stream;
 
-public record PathLookup(
-    Path homeDir,
-    Path configDir,
-    Path[] dataDirs,
-    Path[] sharedRepoDirs,
-    Path tempDir,
-    Function<String, Stream<String>> settingResolver
-) {}
+// TODO: (jack) new a need interface, but can still be a record
```

Comment:
The todo seems like it can be removed, but a simple javadoc would be good

Reply:
Added.